src/main/groovy/com/orctom/gradle/archetype/util/FileUtils.groovy

```diff
@@ -1,73 +1,176 @@
 package com.orctom.gradle.archetype.util
 
+import com.orctom.gradle.archetype.ConflictResolutionStrategy
 import groovy.io.FileType
 import groovy.text.GStringTemplateEngine
+import org.gradle.api.logging.Logger
+import org.gradle.api.logging.Logging
 
 import java.nio.file.Files
+import java.nio.file.StandardCopyOption
+
+import static com.orctom.gradle.archetype.ConflictResolutionStrategy.FAIL
+import static com.orctom.gradle.archetype.ConflictResolutionStrategy.OVERWRITE
 
 class FileUtils {
 
-  def static engine = new GStringTemplateEngine()
+  static final Logger log = Logging.getLogger(getClass().name)
+  static engine = new GStringTemplateEngine()
 
+  /** Provides collection of all files from the source directory.
+   *
+   * @param sourceDir source directory, i.e. directory with the templates(s)
+   * @return list of all files in the source directory, including sub-directories
+   */
   static List<File> getTemplates(File sourceDir) {
-    List<File> list = []
+    log.info("source dir: '{}'", sourceDir.name)
+
+    List<File> sourceFiles = []
     sourceDir.eachFileRecurse(FileType.ANY) { file ->
-      list << file
+      sourceFiles << file
     }
-    list
+
+    sourceFiles
+  }
+
+  /** Provides collection of all non-templates files from the source directory.
+   *
+   * @param sourceDir source directory, i.e. directory with the templates(s)
+   * @param defaultNonTemplatesFile location of the default non-templates files
+   *
+   * @return set of all non-templates files in the source directory, including sub-directories
+   */
+  static Set<String> getNonTemplates(defaultNonTemplatesFile, sourceDir) {
+    Set<String> nonTemplates = []
+
+    // TODO: allow template-specific settings overriding the default
```

Comment:
I mean that what is no-template could be overriden on template level, to have some defaults and to be able to extend them - developers in team can have different  access levels, so this can be really helpful in situations when a developer working on template has not write access to the default list of non-templates.